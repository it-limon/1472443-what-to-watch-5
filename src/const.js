export const MAX_COMMENTS_PER_COLUMN = 2;
export const MAX_GENRES_COUNT = 9;
export const SIMILAR_MOVIES_COUNT = 4;
export const DEFAULT_SHOWN_MOVIES_COUNT = 8;
export const DEFAULT_REVIEW_RATING = 3;
export const MIN_COMMENT_LENGTH = 50;
export const MAX_COMMENT_LENGTH = 400;
export const ALL_GENRES = `All genres`;

export const MovieRating = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very Good`,
  AWESOME: `Awesome`
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`
};

export const AppPages = {
  MAIN: `MAIN`,
  AUTH: `AUTH`,
  MOVIE: `MOVIE`,
  MYLIST: `MYLIST`,
  REVIEW: `REVIEW`,
  PLAYER: `PLAYER`
};

export const AppRoute = {
  MAIN: `/`,
  LOGIN: `/login`,
  FILMS: `/films`,
  MYLIST: `/mylist`,
  REVIEW: `/review`,
  PLAYER: `/player`
};

export const APIRoute = {
  FILMS: `/films`,
  FAVORITE_FILMS: `/favorite`,
  PROMO_MOVIE: `/films/promo`,
  COMMENTS: `/comments`,
  LOGIN: `/login`
};
