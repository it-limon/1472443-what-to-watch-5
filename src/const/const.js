export const MovieRatings = [`Bad`, `Normal`, `Good`, `Very Good`, `Awesome`];
export const REVIEW_COLUMNS_COUNT = 2;
export const LIKE_THIS_MOVIE_COUNT = 4;
