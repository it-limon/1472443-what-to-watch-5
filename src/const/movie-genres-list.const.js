export const MovieGenresList = [
  {
    key: 0,
    ord: 0,
    name: `All genres`,
    filterName: `All genres`,
  },
  {
    key: 1,
    ord: 1,
    name: `Comedie`,
    filterName: `Comedies`,
  },
  {
    key: 2,
    ord: 2,
    name: `Crime`,
    filterName: `Crime`,
  },
  {
    key: 3,
    ord: 3,
    name: `Documentary`,
    filterName: `Documentary`,
  },
  {
    key: 4,
    ord: 4,
    name: `Drama`,
    filterName: `Dramas`,
  },
  {
    key: 5,
    ord: 5,
    name: `Horror`,
    filterName: `Horror`,
  },
  {
    key: 6,
    ord: 6,
    name: `Kids & Family`,
    filterName: `Kids & Family`,
  },
  {
    key: 7,
    ord: 7,
    name: `Romance`,
    filterName: `Romance`,
  },
  {
    key: 8,
    ord: 8,
    name: `Sci-Fi`,
    filterName: `Sci-Fi`,
  },
  {
    key: 9,
    ord: 9,
    name: `Thriller`,
    filterName: `Thrillers`,
  }
];
