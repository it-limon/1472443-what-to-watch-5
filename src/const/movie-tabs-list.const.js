export const MovieTabsList = [
  {
    key: 0,
    name: `Overview`
  },
  {
    key: 1,
    name: `Details`
  },
  {
    key: 2,
    name: `Reviews`
  }
];
