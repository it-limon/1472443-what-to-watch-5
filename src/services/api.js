import axios from "axios";
import {toStandardKeys} from "../utils";

const BACKEND_URL = `https://5.react.pages.academy/wtw`;
const REQUEST_TIMEOUT = 5000;

const HttpCode = {
  UNAUTHORIZED: 401,
  PAGE_NOT_FOUND: 404
};

export const createAPI = (onUnauthorized, onPageNotFound) => {
  const api = axios.create({
    baseURL: BACKEND_URL,
    timeout: REQUEST_TIMEOUT,
    withCredentials: true
  });

  const onSuccess = (response) => {
    onPageNotFound(false);
    response.data = toStandardKeys(response.data);
    return response;
  };

  const onFail = (err) => {
    const {response} = err;

    if (!response) {
      onPageNotFound(true);
    }

    if (response.status === HttpCode.UNAUTHORIZED) {
      onUnauthorized();
    }

    if (response.status === HttpCode.PAGE_NOT_FOUND) {
      onPageNotFound(true);
    }

    throw err;
  };

  api.interceptors.response.use(onSuccess, onFail);

  return api;
};
