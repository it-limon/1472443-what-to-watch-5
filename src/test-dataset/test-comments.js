export const testComments = [
  {
    id: 1,
    user: {
      id: 1,
      name: `User 1`
    },
    rating: 8,
    comment: `Comment 1`,
    date: `1995-12-17T03:24:00`
  },
  {
    id: 2,
    user: {
      id: 2,
      name: `User 2`
    },
    rating: 7,
    comment: `Comment 2`,
    date: `1996-12-17T03:24:00`
  },
  {
    id: 3,
    user: {
      id: 3,
      name: `User 3`
    },
    rating: 5,
    comment: `Comment 3`,
    date: `1997-12-17T03:24:00`
  }
];