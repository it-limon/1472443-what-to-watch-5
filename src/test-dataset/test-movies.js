export const testMovies = [
  {
    id: 1,
    name: `El Camino: A Breaking Bad Movie`,
    posterImage: `https://cdn.pixabay.com/photo/2020/09/27/04/15/cat-5605615_960_720.jpg`,
    previewImage: `https://cdn.pixabay.com/photo/2020/11/16/16/13/trees-5749406_960_720.jpg`,
    backgroundImage: `https://cdn.pixabay.com/photo/2020/11/16/16/05/hoverfly-5749361_960_720.jpg`,
    backgroundColor: `#000000`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    description: `Fugitive Jesse Pinkman runs from his captors, the law, and his past.`,
    rating: 7.2,
    scoresCount: 36346,
    director: `Vince Gilligan`,
    starring: [`Aaron Paul`, `Jonathan Banks`, `Matt Jones`],
    runTime: 122,
    genre: `Drama`,
    released: 2019,
    isFavorite: false
  },
  {
    id: 2,
    name: `The Dark Knight`,
    posterImage: `https://cdn.pixabay.com/photo/2020/09/27/04/15/cat-5605615_960_720.jpg`,
    previewImage: `https://cdn.pixabay.com/photo/2020/11/16/16/13/trees-5749406_960_720.jpg`,
    backgroundImage: `https://cdn.pixabay.com/photo/2020/11/16/16/05/hoverfly-5749361_960_720.jpg`,
    backgroundColor: `#000000`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    description: `When the menace known as the Joker wreaks havoc and chaos on the people of Gotham, 
                  Batman must accept one of the greatest psychological and physical tests of his ability 
                  to fight injustice.`,
    rating: 9,
    scoresCount: 2271245,
    director: `Christopher Nolan`,
    starring: [`Christian Bale`, `Heath Ledger`, `Aaron Eckhart`],
    runTime: 152,
    genre: `Crime`,
    released: 2008,
    isFavorite: true
  }
];