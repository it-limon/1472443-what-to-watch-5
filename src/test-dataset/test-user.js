export const testUser = {
  id: 1,
  email: `user@mail.ru`,
  name: `User Name`,
  avatarUrl: `https://cdn.pixabay.com/photo/2017/06/23/08/13/china-2433891_960_720.png`
};