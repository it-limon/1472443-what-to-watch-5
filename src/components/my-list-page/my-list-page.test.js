import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import MyListPage from "./my-list-page";
import {Provider} from "react-redux";
import {AuthorizationStatus} from "../../const";
import {Route, BrowserRouter} from "react-router-dom";
import {createAPI} from "../../services/api";
import thunk from "redux-thunk";
import {testMovies} from "../../test-dataset/test-movies";
import {testUser} from "../../test-dataset/test-user";

const noop = () => {};

let mockStore = null;
let store = null;

const api = createAPI(() => {}, () => {});
let middlewares = [thunk.withExtraArgument(api)];

describe(`Render MyListPage`, () => {

  it(`Render MyListPage with 2 favorite movies`, () => {
    mockStore = configureStore(middlewares);
    store = mockStore({
      DATA: {
        favoriteMovies: testMovies
      },
      USER: {
        authorizationStatus: AuthorizationStatus.AUTH,
        userInfo: testUser
      }
    });

    const tree = renderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <Route>
              <MyListPage
                onLoadFavoriteMoviesList={noop}
              />
            </Route>
          </BrowserRouter>
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Render MyListPage without favorite movies`, () => {
    mockStore = configureStore(middlewares);
    store = mockStore({
      DATA: {
        favoriteMovies: []
      },
      USER: {
        authorizationStatus: AuthorizationStatus.AUTH,
        userInfo: testUser
      }
    });

    const tree = renderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <Route>
              <MyListPage
                onLoadFavoriteMoviesList={noop}
              />
            </Route>
          </BrowserRouter>
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

});
