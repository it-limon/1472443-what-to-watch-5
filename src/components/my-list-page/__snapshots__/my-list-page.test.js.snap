// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MyListPage Render MyListPage with 2 favorite movies 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <div
        className="user-block__avatar"
      >
        <img
          alt="User Name"
          height="63"
          src="https://cdn.pixabay.com/photo/2017/06/23/08/13/china-2433891_960_720.png"
          width="63"
        />
      </div>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            className="player__video"
            muted={true}
            poster="https://cdn.pixabay.com/photo/2020/11/16/16/13/trees-5749406_960_720.jpg"
            src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="#"
            onClick={[Function]}
          >
            El Camino: A Breaking Bad Movie
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            className="player__video"
            muted={true}
            poster="https://cdn.pixabay.com/photo/2020/11/16/16/13/trees-5749406_960_720.jpg"
            src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="#"
            onClick={[Function]}
          >
            The Dark Knight
          </a>
        </h3>
      </article>
    </div>
  </section>
  <footer>
    <a
      className="logo__link logo__link--light"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
    <div
      className="copyright"
    >
      <p>
        © 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;

exports[`Render MyListPage Render MyListPage without favorite movies 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <div
        className="user-block__avatar"
      >
        <img
          alt="User Name"
          height="63"
          src="https://cdn.pixabay.com/photo/2017/06/23/08/13/china-2433891_960_720.png"
          width="63"
        />
      </div>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    />
  </section>
  <footer>
    <a
      className="logo__link logo__link--light"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
    <div
      className="copyright"
    >
      <p>
        © 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
